util/chplenv/chpl_comm.py

```diff
@@ -10,17 +11,27 @@ def get():
     comm_val = overrides.get('CHPL_COMM')
     if not comm_val:
         platform_val = chpl_platform.get('target')
+        # use gasnet for ibv
+        if shutil.which('ibstat'):
```

Comment:
It is interesting to put this first in the if/elsif chain. IMO we should put the platforms where we know the network (XC, EX) first. Then we should check for ibstat vs cxi_stat, and finally have the defaults for platforms where it could be either (cs, apollo, xd). Of course there are other ways to write that.

Reply:
We discussed this offline, and the latest commit, https://github.com/chapel-lang/chapel/pull/26921/commits/dfacd5bd4611fc75a102f9d0a73054b85cab9539, addresses these concerns by adding a new `get_network` method that is then used to help deduce the platform, the comm value, substrate, etc